hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/search/ExtendedHSearchResourceProjection.java

```diff
@@ -29,12 +29,14 @@
  * Query result when fetching full resources from HSearch.
  */
 public class ExtendedHSearchResourceProjection {
+	public static final String RESOURCE_NOT_STORED_ERROR = "Resource not stored in search index: ";
+
 	final long myPid;
 	final String myForcedId;
 	final String myResourceString;
 
 	public ExtendedHSearchResourceProjection(long thePid, String theForcedId, String theResourceString) {
-		Validate.notEmpty(theResourceString, "Resource not stored in search index: " + thePid);
+		Validate.notEmpty(theResourceString, RESOURCE_NOT_STORED_ERROR + thePid);
```

Comment:
FYI: The `Validate` methods also take a format string with placeholders to avoid generating garbage on the happy path.
```suggestion
		Validate.notEmpty(theResourceString, "Resource not stored in search index: %s", thePid);
```